website_sale_coupon_page/security/ir.model.access.csv

```diff
@@ -0,0 +1,2 @@
+id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
+access_program_user,"Access user to programs",model_sale_coupon_program,,1,0,0,0
```

Comment:
Never give a permission without group. This is a total leak. Normal technique is to add 2 lines for public and portal users, but also record rules should be in place for restricting to published elements. Anyways, as the risk for leaking some data for promotions (like number of sales records, or the rules), we are not going to do this, and put `sudo` on the controller, so the only way to access this information is through that controller.